server/get_header.go

```diff
@@ -189,3 +254,89 @@ func (m *BoostService) getHeader(log *logrus.Entry, ua UserAgent, slot phase0.Sl
 	result.relays = relays[BlockHashHex(result.bidInfo.blockHash.String())]
 	return result, nil
 }
+
+// decodeBid decodes a bid by SSZ or JSON, depending on the provided respContentType
+func decodeBid(respBytes []byte, respContentType, ethConsensusVersion string, bid *builderSpec.VersionedSignedBuilderBid) error {
+	switch respContentType {
+	case MediaTypeOctetStream:
+		if ethConsensusVersion != "" {
+			// Do SSZ decoding
+			switch ethConsensusVersion {
+			case "bellatrix":
+				bid.Version = spec.DataVersionBellatrix
+				bid.Bellatrix = new(builderApiBellatrix.SignedBuilderBid)
+				return bid.Bellatrix.UnmarshalSSZ(respBytes)
+			case "capella":
+				bid.Version = spec.DataVersionCapella
+				bid.Capella = new(builderApiCapella.SignedBuilderBid)
+				return bid.Capella.UnmarshalSSZ(respBytes)
+			case "deneb":
+				bid.Version = spec.DataVersionDeneb
+				bid.Deneb = new(builderApiDeneb.SignedBuilderBid)
+				return bid.Deneb.UnmarshalSSZ(respBytes)
+			case "electra":
+				bid.Version = spec.DataVersionElectra
+				bid.Electra = new(builderApiElectra.SignedBuilderBid)
+				return bid.Electra.UnmarshalSSZ(respBytes)
+			default:
+				return errInvalidForkVersion
+			}
+		} else {
+			return types.ErrMissingEthConsensusVersion
+		}
+	case MediaTypeJSON:
+		// Do JSON decoding
+		return json.Unmarshal(respBytes, bid)
+	}
+	return types.ErrInvalidContentType
+}
+
+// respondGetHeaderJSON responds to the proposer in JSON
+func (m *BoostService) respondGetHeaderJSON(w http.ResponseWriter, result *bidResp) {
+	w.Header().Set("Content-Type", MediaTypeJSON)
+	w.WriteHeader(http.StatusOK)
+
+	// Serialize and write the data
+	if err := json.NewEncoder(w).Encode(&result.response); err != nil {
+		m.log.WithField("response", result.response).WithError(err).Error("could not write OK response")
+		http.Error(w, "", http.StatusInternalServerError)
+	}
+}
+
+// respondGetHeaderSSZ responds to the proposer in SSZ
+func (m *BoostService) respondGetHeaderSSZ(w http.ResponseWriter, result *bidResp) {
+	// Serialize the response
+	var err error
+	var sszData []byte
+	switch result.response.Version {
+	case spec.DataVersionBellatrix:
+		w.Header().Set("Eth-Consensus-Version", "bellatrix")
```

Comment:
consts for the consensus version would be better than hardcoded strings

Reply:
I decided to do this for more of these hardcoded strings.